repos/fdbt-netex-output/src/netex-convertor/netexGenerator.ts

```diff
@@ -370,12 +378,22 @@ const netexGenerator = async (ticket: Ticket, operatorData: Operator[]): Promise
         tariff.fareStructureElements.FareStructureElement = fareStructuresElements;
 
         if (isFlatFareType(ticket)) {
+            const isPricedByDistance = 'pricingByDistance' in ticket.products[0];
```

Comment:
we can move this up

Reply:
But we only want to check it if its a flat fare ticket.